Spike: Resolve SPARQL link query

jonrandahl opened the pull request:
This PR is primarily to update the internal gem versions pertaining to the UKHPI ticket https://github.com/epimorphics/ukhpi/issues/493 as well as enhances the development and deployment workflows through configuration and automation.

- Introduces development environment configurations via `.env.development`
- Automates Docker image building on issue/spike/task branch commits using a `.githooks/post-commit` script
- Restricts pre-commit/pre-push hooks on specific branches
- Ignores local environment files in `.gitignore`
- Makes the Makefile region-aware by using the `AWS_REGION` variable
- Updates gem versions
- Adds `Procfile.dev` for local development using foreman

jonrandahl committed:
Feat: Adds development environment configuration

Sets up the development environment with:

- API service URL configuration
- Port configuration
- Sentry configuration (disabled by default in development)
- Updates the .gitignore to ignore environment files except for the development one
- Updates Makefile to display the starting server message on the console
- Adds a Procfile for running the application in development

jonrandahl committed:
Fix: Uses AWS_REGION variable for ECR

Sets the ECR variable definition to use the AWS_REGION environment variable.

jonrandahl committed:
Refactor: Improves git hook logic

Refactors git hooks to skip checks on hotfix, rebase, and production branches.

* Adds a post-commit hook to build a docker image after a commit
* Updates pre-commit and pre-push hooks to check branch names before running.

jonrandahl committed:
build: Updates internal gems to latest versions

*   Bumps json-rails-logger to address logging enhancements.
*   Updates lr_common_styles for styling improvements.
*   Updates qonsole_rails to include query improvements.

jonrandahl committed:
Bump: Updates version to 2.0.6

Increments the revision number cadence

jonrandahl committed:
docs: updated CHANGELOG for v2.0.6

Enhances the development and deployment workflows through configuration and automation.

- Introduces development environment configurations via `.env.development`
- Automates Docker image building on issue/spike/task branch commits using a `.githooks/post-commit` script
- Restricts pre-commit/pre-push hooks on specific branches
- Ignores local environment files in `.gitignore`
- Makes the Makefile region-aware by using the `AWS_REGION` variable
- Updates gem versions
- Adds `Procfile.dev` for local development using foreman

bogdanadrianmarc reviewed (approved):
The build is failing, otherwise lgtm 👍

jonrandahl committed:
Build: Updates to terser for ES6+ support

Updates the JavaScript compressor from `uglifier` to `terser` to better support modern JavaScript (ES6+) syntax.

jonrandahl committed:
Build: Replaces uglifier with terser

Replaces the JavaScript compressor to use terser.

- Improves build performance by switching to a more efficient compressor.
- Comments out the previous compressor.
